### ISSUE TITLE
do not skip TestVisualizationImplementation

### ISSUE THREAD

**Issue by 1ucian0** [Qiskit/qiskit-terra#1073]
There is a `TODO` in `test_visualization_output.py`.
```
 # TODO: Enable for refactoring purposes and enable by default when we can
 # decide if the backend is available or not.
@unittest.skip('Useful for refactoring purposes, skipping by default.')
class TestVisualizationImplementation(QiskitTestCase):
```
I think is time to remove the `TODO` and start running it. Especially in the context of the new `circuit_drawer` interface.

**Comment by 1ucian0** [comment]
I moved the skip to the specific tests in 6618ed7.

**Comment by ajavadia** [comment]
@1ucian0 i don't get this. what's the point of skipping these (what does it mean by "useful for refactoring purposes"). There are other tests of latex drawer and mpl drawer that we don't skip. I think we should remove the skip.

**Comment by 1ucian0** [comment]
This issue is about stop skipping them. When we had that attempt to have a meta test for the visualizers (https://github.com/Qiskit/qiskit-terra/pull/1617), we could close it. But we reverted it (https://github.com/Qiskit/qiskit-terra/pull/1780). Any way, the idea is to test the visualizers, not just to skip it.

**Comment by 1ucian0** [comment]
In order to close this issue, we need to move the following tests to binder (see https://github.com/Qiskit/qiskit-terra/pull/4544):
```
test//python/visualization/test_circuit_visualization_output.py:83:    @unittest.skip('Useful for refactoring purposes, skipping by default.')
test//python/visualization/test_circuit_visualization_output.py:94:    @unittest.skip('Useful for refactoring purposes, skipping by default.')
test//python/visualization/visualization.py:16:Useful for refactoring purposes."""
test//python/visualization/test_pulse_visualization_output.py:83:    @unittest.skip('Useful for refactoring purposes, skipping by default.')
test//python/visualization/test_pulse_visualization_output.py:94:    @unittest.skip('Useful for refactoring purposes, skipping by default.')
test//python/visualization/test_pulse_visualization_output.py:109:    @unittest.skip('Useful for refactoring purposes, skipping by default.')
test//python/visualization/test_pulse_visualization_output.py:120:    @unittest.skip('Useful for refactoring purposes, skipping by default.')
test//python/visualization/test_pulse_visualization_output.py:131:    @unittest.skip('Useful for refactoring purposes, skipping by default.')
test//python/visualization/test_pulse_visualization_output.py:142:    @unittest.skip('Useful for refactoring purposes, skipping by default.')
test//python/visualization/test_pulse_visualization_output.py:153:    @unittest.skip('Useful for refactoring purposes, skipping by default.')
test//python/visualization/test_pulse_visualization_output.py:165:    @unittest.skip('Useful for refactoring purposes, skipping by default.')
```

**Comment by enavarro51** [comment]
@1ucian0 Are these still relevant - `test_circuit_visualization_output.py`, `visualization.py`, and `test_pulse_visualization_output.py`? I believe there are other tests that cover these and the test images are way out of date. Should they be removed along with their .png files?

**Comment by 1ucian0** [comment]
* `test.python.visualization.test_circuit_visualization_output.TestCircuitVisualizationImplementation.test_latex_drawer`: I think this is being handled by https://github.com/Qiskit/qiskit-terra/pull/6450 . If so, once 6450 is merged, it can be removed.
 * `test.python.visualization.test_circuit_visualization_output.TestCircuitVisualizationImplementation.test_matplotlib_drawer`: This one is probably already covered by the current mpl snapshot tests.
 * Those in `test/python/visualization/test_pulse_visualization_output` they need to be moved to the #4544 style, with updated references.
 * I think `test/python/visualization/visualization.py` can be removed, since it does image comparison that we are not using anymore.

**Comment by enavarro51** [comment]
First off, since most all of these tests are skipped and have been skipped for a couple of years, I'm not sure it's of much value to port any of these. It seems more reasonable to delete all this and open issues for places where testing is inadequate, like #2307.
- Again, for the latex test, since this is already skipped, why wait for #6450.
- Agreed on the mpl test
- If there's a plan to port pulse testing to #4544 style, there should be a new issue for that, and I assume the current tests are inadequate for porting, given they're old and there have been a lot of pulse changes.
- Agreed on the old mpl testing in visualization.py.

**Comment by 1ucian0** [comment]
Removing the skipped tests and create issue for non covered code would be a way to close this issue indeed.

**Comment by enavarro51** [comment]
Ok, great. I can clean this up.
Is there a plan in place to move pulse testing to the #4544 model? If so, I'll open a new issue for that.
